lib/Dialect/Torch/Transforms/ReduceOpVariants.cpp

```diff
@@ -61,8 +69,13 @@ class ConvertHasValueSemanticsOpsToValueTensors : public RewritePattern {
       : RewritePattern(MatchAnyOpTypeTag(), /*benefit=*/1, context) {}
   LogicalResult matchAndRewrite(Operation *op,
                                 PatternRewriter &rewriter) const override {
-    if (!op->hasTrait<Torch::OpTrait::HasValueSemantics>())
+    if (isa<OperatorOp>(op)) {
+      if (!operatorOpHasValueSemantics(cast<OperatorOp>(op))) {
```

Comment:
you need a nesting here because you want the outer branch to "succeed" for appropriate instances - if `isa<OperatorOp>(op)` and `operatorOpHasValueSemantics(cast<OperatorOp>(op))` then succeed.